jina/peapods/peas/__init__.py

```diff
@@ -311,13 +311,20 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         self.close()
 
     def _get_runtime_cls(self) -> Tuple[Any, bool]:
-        is_remote_controlled = False
-        if self.args.host != __default_host__ and not self.args.disable_remote:
+        gateway_runtime_dict = {
+            GatewayProtocolType.GRPC: 'GRPCRuntime',
+            GatewayProtocolType.WEBSOCKET: 'WebSocketRuntime',
+            GatewayProtocolType.HTTP: 'HTTPRuntime',
+        }
+        if (
```

Comment:
This is a highly non-obvious fix of the bug. Why does a simple
```
disable_remote = self.args.hasattr('disable_remote') and self.args.disable_remote
if self.args.host != __default_host__ and not disable_remote:
    ...
```
not work?

Reply:
Because it is not just about `disable_remote`. The fact that, `disable_remote` is _not set only_ for gateway peas, lies in a different part of the code.  Even if that is checked properly, we should explicitly avoid setting `JinadRuntime` for gateway peas.